Dynamic UI docs (#342)
RoadieHQ/marketing-site: content/docs/getting-started/updating-the-ui/index.md

---
title: Updating the Backstage UI
lastUpdated: '2021-08-06T12:00:00.0Z'
description: How to configure the Backstage UI Layout
---

## Introduction

Roadie allows you to configure which plugins you want to use via our UI. This guide explains the different types of
plugin components and the different places they can be integrated into the app.

## Prerequisites

> * You must be an admin to edit the UI layout. Please complete [this guide](/docs/getting-started/getting-started-for-admins/) before moving on to this one.

Reviewer: This doesn't need to be a bullet, since there is just one bullet point.

```suggestion
You must be an admin to edit the UI layout. Please complete [this guide](/docs/getting-started/getting-started-for-admins/) before moving on to this one.
```

## About Plugins

Backstage is a plugin based application where each plugin provides a certain piece of functionality. A plugin may
integrate with some external system such as GitHub to display information for a certain Entity e.g. Show pull request
information for a particular service. 

You can read more about Backstage plugins [here](https://backstage.io/docs/plugins/).

> In terms of the Backstage layout plugins can provide [three types of component](https://backstage.io/docs/plugins/composability#naming-patterns):

Reviewer 2: ```suggestion
In terms of the Backstage layout plugins can provide [three types of components](https://backstage.io/docs/plugins/composability#naming-patterns):
```

Reviewer: I would try to avoid using the word "component" since it already applies to a formal Backstage concept. How about "format" or "widget"?

```suggestion
Backstage plugins can provide UI elements in [three different formats](https://backstage.io/docs/plugins/composability#naming-patterns):
```

Reviewer: @iain-b I think you might have forgotten to push a change here? The text appears unchanged.

Author: Nevermind I just amended the commit

Reviewer: Yeah I guess. The screenshots are still dark mode on that branch but you said that was fixed in standup.

I'm confused now 😬 .

Author: Man this has been a nightmare... 😓  They're there now.

1. Card - These are tiles which can be arranged into dashboards in the software catalog. E.g. The About card.

> 2. Content - These are tabs which can be integrated with the catalog. E.g. The techdocs tab.

Reviewer: TechDocs should be capitalized like it is in [the backstage.io docs](https://backstage.io/docs/features/techdocs/techdocs-overview).

```suggestion
2. Content - These are tabs which can be integrated with the catalog. E.g. The TechDocs tab.
```

3. Page - These are top-level pages in Backstage, outside of the catalog. Accessible via links in the sidebar. E.g. The catalog itself is a Page.

## Updating the Catalog Layout

Admin users can add components to catalog pages to configure what is displayed to users when they view an entity. Layouts are configured
per entity **kind** except for Component entities which have layouts per **type**. For example, "User" and "Group" are each kinds
of entity and so User entities have a different layout to Group entities. Similarly, Component entities have types such
as "Service", "Library", "Website", etc and different types can have different layouts. However, a particular entity can't
have its own layout; A particular Group or service can't have a unique layout. This is to encourage a consistent UX.

### Updating Tabs

To add a tab click the plus icon at the end of the tab list.

> ![Add tab button](./add-tab-button.png)

Reviewer: I realize this is going to be a pain in the ass, but could you switch the screenshots to light mode please? 😳 All other screenshots in the docs, and the roadie.io website itself, are light mode.

Sorry!

Then select the component to add via the autocomplete and give the tab a display name and click "Create".

![Add tab dialog](./add-tab-screen.png)

To rearrange or delete tabs select the "pencil" edit icon on the far right of the tab bar.

![Edit tab layout](./edit-tab-layout.png)

Then drag and drop to rearrange or click the "x" icon to remove a tab then *click the save icon* on the right hand side.

![Move tab](./move-tab.png)

### Updating Dashboards

To edit a dashboard click the cog icon on the top right of the dashboard below the tab bar.

![Edit Dashboard Button](./edit-grid-button.png)

> This will switch the dashboard into and edit view. You can drag and drop to rearrange or click the dustbin icon to remove a card. 

Reviewer 2: ```suggestion
This will switch the dashboard into an edit view. You can drag and drop to rearrange or click the dustbin icon to remove a card. 
```
